### PR TITLE
Motion planner refactor

### PR DESCRIPTION
**Pull request by asaba96**
Created test abstract to unit test HLM rewrite and transition checking.

### CODE REVIEW

File: scripts/high_level_motion_unit_test.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+import sys
+import rospy
+from iarc7_msgs.msg import OdometryArray
+import actionlib
+import tf2_ros
+from iarc7_motion.msg import QuadMoveGoal, QuadMoveAction
+from iarc7_safety.SafetyClient import SafetyClient
+
+def test():
+    safety_client = SafetyClient('high_level_motion_unit_test')
```
Comment (amiller27) [inline scripts/high_level_motion_unit_test.py]: Name should get changed here

File: scripts/high_level_motion_unit_test.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+    # Waits until the action server has started up and started
+    # listening for goals.
+    client.wait_for_server()
+
+    rospy.sleep(2.0)
+
+    # change element in array to test diff roombas
+    roomba_id = roomba_array.data[5].child_frame_id 
+    roomba_id = roomba_id [0:len(roomba_id)-10]
```
Comment (amiller27) [inline scripts/high_level_motion_unit_test.py]: No space before the `[` please, also please comment why there's a `10` here

File: scripts/high_level_motion_unit_test.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    client = actionlib.SimpleActionClient("motion_planner_server", QuadMoveAction)
+
+    # Waits until the action server has started up and started
+    # listening for goals.
+    client.wait_for_server()
+
+    rospy.sleep(2.0)
+
+    # change element in array to test diff roombas
+    roomba_id = roomba_array.data[5].child_frame_id
```
Comment (amiller27) [inline scripts/high_level_motion_unit_test.py]: What's `5`?

File: scripts/high_level_motion_unit_test.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+import tf2_ros
+from iarc7_motion.msg import QuadMoveGoal, QuadMoveAction
+from iarc7_safety.SafetyClient import SafetyClient
+
+def test():
+    safety_client = SafetyClient('high_level_motion_unit_test')
+    # Since this abstract is top level in the control chain there is no need to check
+    # for a safety state. We can also get away with not checking for a fatal state since
+    # all nodes below will shut down.
+    assert(safety_client.form_bond())
```
Comment (amiller27) [inline scripts/high_level_motion_unit_test.py]: No parentheses here, just `assert safety_client.form_bond()`, and same below.  I know there are parentheses after asserts all over this codebase, but there shouldn't be

File: scripts/high_level_motion_unit_test.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    try:
+        roomba_array = []
+        # Initializes a rospy node so that the SimpleActionClient can
+        # publish and subscribe over ROS.
+        _roomba_status_sub = rospy.Subscriber('roombas', 
+                         OdometryArray, _receive_roomba_status)
+
+        rospy.init_node('high_level_motion_unit_test')
+        test()
+        while not rospy.is_shutdown():
```
Comment (amiller27) [inline scripts/high_level_motion_unit_test.py]: Just call `rospy.spin` here instead of this loop

Comment (asaba96) [reply scripts/high_level_motion_unit_test.py]: fixed

File: scripts/high_level_motion_unit_test.py (hunk 2 above)
Comment (asaba96) [reply scripts/high_level_motion_unit_test.py]: Fixed

File: scripts/high_level_motion_unit_test.py (hunk 3 above)
Comment (asaba96) [reply scripts/high_level_motion_unit_test.py]: fixed, arbitrary roomba

File: scripts/high_level_motion_unit_test.py (hunk 4 above)
Comment (asaba96) [reply scripts/high_level_motion_unit_test.py]: fixed

File: scripts/high_level_motion_unit_test.py (hunk 1 above)
Comment (asaba96) [reply scripts/high_level_motion_unit_test.py]: fixed, changed to `invalid_task_transition_unit_test`

**Review by amiller27 — APPROVED** [review]

**Pull request merged**